UI/src/view/point_asset/servicePointLabel.js

```diff
@@ -0,0 +1,160 @@
+      }
+
+      function getHeight() {
+        return labelProperty && labelProperty.height ? labelProperty.height : 35;
+      }
+
+      return {
+        findImage: findImage,
+        getTextOffsetX: getTextOffsetX,
+        getTextOffsetY: getTextOffsetY,
+        getValidation: getValidation,
+        getValue : getValue,
+        getUnit : getUnit,
+        getAdditionalInfo: getAdditionalInfo,
+        getMaxLength: getMaxLength,
+        getHeight: getHeight
+      };
+    };
+
+    var validateText = function () { return true;};
+
+    this.renderGroupedFeatures = function(assets, zoomLevel, getPoint){
+      if(!this.isVisibleZoom(zoomLevel))
+        return [];
+      var groupedAssets = me.getGroupedFeatures(assets, zoomLevel);
+      return _.flatten(_.chain(groupedAssets).map(function(assets){
+        var imgPosition = {x: 0 , y: me.stickPosition.y};
+        return _.map(assets, function(asset){
+          var values = me.getValue(asset);
+          if (!(_.isUndefined(values) || _.isEmpty(values))) {
+            var styles = [];
+            styles = styles.concat(me.getStickStyle());
+            styles = styles.concat(_.flatMap(_.map(values, function(value) {
+              imgPosition.y += me.getLabelProperty(value).getHeight();
+              return me.getStyle(value, imgPosition );
+            })));
+            var feature = me.createFeature(getPoint(asset));
+            feature.setStyle(styles);
+            feature.setProperties(_.omit(asset, 'geometry'));
+            return feature;
+          }
+        });
+      }).filter(function(feature){ return !_.isUndefined(feature); }).value());
+    };
+
+
+    var defaultInfoValues =
+      [ {type: 5, text: 'Rajanylityspaikka', colorText: '#ffffff'},
+        {type: 13, text: 'Lastausterminaali', colorText: '#000000'},
+        {type: 14, text: 'Pysäköintialue', colorText: '#000000'},
+        {type: 17, text: 'Latauspiste', colorText: '#ffffff'}];
+
+    function getDefaultColorText(type) {
+      var color =_.find(defaultInfoValues, function(defaultInfo) {return defaultInfo.type === type;});
+      return color ? color.colorText : "";
+    }
+
+    function getDefaultText(type) {
+     var text =_.find(defaultInfoValues, function(defaultInfo) {return defaultInfo.type === type;});
+      return text ? text.text : "";
+    }
+
+    me.getValue = function (asset) {
+      return _.map(asset.services, function (service) {
+        return {value: getDefaultText(service.serviceType), type: service.serviceType, typeExtension: service.typeExtension, textColor: getDefaultColorText(service.serviceType)};
+      });
+    };
+
+    me.getStyle = function (trafficSign, position) {
+      return [me.backgroundStyle(trafficSign, position), new ol.style.Style({
+        text: new ol.style.Text({
+          text: this.textStyle(trafficSign),
+          fill: new ol.style.Fill({
+            color: trafficSign.textColor
```

Comment:
check if it possible use the main method (father) override only the trafficSign.textColor (sign.textColor)